staging/src/k8s.io/apiserver/pkg/endpoints/request/context.go

```diff
@@ -94,3 +95,22 @@ func AuditEventFrom(ctx context.Context) *audit.Event {
 	ev, _ := ctx.Value(auditKey).(*audit.Event)
 	return ev
 }
+
+// WithSecondaryDeadline creates a child context with a deadline specified in timeout
+func WithSecondaryDeadline(parent context.Context, timeout time.Duration) (context.Context, context.CancelFunc) {
+	if timeout <= 0 {
+		return parent, func() {}
+	}
+
+	deadline, hasDeadline := parent.Deadline()
+	if !hasDeadline {
+		return context.WithTimeout(parent, timeout)
+	}
+
+	remaining := time.Until(deadline)
+	if remaining <= timeout {
```

Comment:
I thought `WithTimeout` called `WithDeadline`, which I thought already handled the "set to a value that's longer" case https://golang.org/src/context/context.go?s=13362:13430#L420

Reply:
Yes, you are right. I overlooked this. I removed the function.